### PR TITLE
libobs: Update to SIMDe 0.7.0 via a single file

### PR DESCRIPTION
**Pull request by mr-c**
### Description

libobs: Update to SIMDe 0.7.0 via a single file

Via https://github.com/simd-everywhere/simde/blob/c3d7abfaba6729a8b11d09a314b34a4db628911d/amalgamate.py
run against https://github.com/simd-everywhere/simde/commit/f68981de04072012dcc888716dedae2a345d0e45

Simplify usage of the SIMDe header

This obviates the need for sse2neon as well and fixes compilation of all files that referenced sse-intrin.h on all architectures, not just arm*.

### Motivation and Context

Since the [last update](https://github.com/obsproject/obs-studio/pull/3006) (sometime between SIMDe v0.5.0 and v0.6.0) there have been many optimizations and fixes leading up to the recent [SIMDe v0.7.0](https://github.com/simd-everywhere/simde/releases/tag/v0.7.0) release.

### How Has This Been Tested?
I tested this by building the [Debian package for https://github.com/obsproject/obs-studio/releases/tag/26.0.2 with the equivalent of these changes](https://tracker.debian.org/news/1204556/accepted-obs-studio-2602dfsg1-110exp0simde-source-into-experimental/) on amd64 and arm64.

### Types of changes
- Performance enhancement (non-breaking change which improves efficiency) 
- Code cleanup (non-breaking change which makes code smaller or more readable) 

### Checklist:

- [x] My code has been run through [clang-format](https://github.com/obsproject/obs-studio/blob/master/.clang-format).
- [x] I have read the [**contributing** document](https://github.com/obsproject/obs-studio/blob/master/CONTRIBUTING.rst).
- [x] My code is not on the master branch.
- [x] The code has been tested.
- [x] All commit messages are properly formatted and commits squashed where appropriate.
- [x] I have included updates to all appropriate documentation.

### CODE REVIEW

**Comment by mr-c** [comment]
Heh, the single-file simde-sse2.h causes `clang-format` to run out of memory, even on my 64GiB machine. 

Shall I except that one file, or switch back to the collection of headers?

File: libobs/CMakeLists.txt [hunk 1]
```diff
@@ -369,7 +357,6 @@ set(libobs_util_SOURCES
 set(libobs_util_HEADERS
 	util/curl/curl-helper.h
 	util/sse-intrin.h
-	util/sse2neon.h
```
Comment (kkartaltepe) [inline libobs/CMakeLists.txt:372]: it removes references to `sse2neon` but doesnt remove the header.

File: libobs/util/simde/README.libobs [hunk 2]
```diff
@@ -1,5 +1,5 @@
-This is a slightly modified version of https://github.com/nemequ/simde/commit/cafec4b952fa5a31a51a10326f97c2e7c9067771
-sse{,2}.h and mmx.h was moved down from the original "x86" subdirectory,
-subsequently the '#include "../simde-common.h"' line in mmx.h was changed to '#include "simde-common.h"'
+This is a slightly modified version of the simde directory in
```
Comment (kkartaltepe) [inline libobs/util/simde/README.libobs:1]: Is the only modification now that unused files have been removed?

File: libobs/CMakeLists.txt (hunk 1 above)
Comment (mr-c) [reply libobs/CMakeLists.txt:372]: Fixed, thanks!

File: libobs/util/simde/README.libobs (hunk 2 above)
Comment (mr-c) [reply libobs/util/simde/README.libobs:1]: Correct, now the files match the SIMDe repo, except for the reformatting by `clang-format`

**Comment by Fenrirthviti** [comment]
I don't see any previous clang-format runs on anything in the util/simde folder, so probably best to just exclude that location entirely as it's being pulled from another project anyway. If we format it, it would make updating or cherry picking any specific fixes that may be required (however unlikely) more annoying.

**Comment by mr-c** [comment]
@Fenrirthviti The [original import](https://github.com/obsproject/obs-studio/pull/2069) used `clang-format` and so my [last update](https://github.com/obsproject/obs-studio/pull/3006/files#diff-f6a0a5ab729a09cd2d3b806ae882a142ee0aeca20a665cfc177d4699a33b633bR5) did so as well.

I'm happy to go with whatever you all decide on :-)

**Pull request merged**